firmware/boron-ins-fsm/src/im21door.cpp

```diff
@@ -153,30 +153,23 @@ void threadBLEScanner(void *param) {
   BLE.setScanTimeout(5);
 
   while(true){
-
-    int count = BLE.scan(scanResults, SCAN_RESULT_MAX);
-
+    //filter has to be remade each time, as globalDoorId (from eeprom) is usually read while this thread is already in while loop
+    BleScanFilter filter; 
+    char address[sizeof("B8:7C:6F:5C:23:49")]; 
+    sprintf(address, "B8:7C:6F:%02X:%02X:%02X", globalDoorID.byte3, globalDoorID.byte2, globalDoorID.byte1); 
+    filter.deviceName("iSensor ").address(address); 
+
+    //scanning for door sensors of correct id
+    Vector<BleScanResult> scanResults = BLE.scanWithFilter(filter); 
+    
     //loop over all devices found in the BLE scan
-    for (int i = 0; i < count; i++) {
+    for (auto scanResult : scanResults) {
```

Comment:
It's been awhile since I've done anything in C++, why would one use `auto` instead of specifying the known `BleScanResult` type of `scanResult`?

Reply:
No good reason to. I have changed it to `BleScanResult` for better readability.